app/__init__.py

```diff
@@ -16,13 +17,20 @@
 
 
 def null_to_dash(value):
+    """Filter that converts string values that are "null" or Nones to dash"""
     if value == "null":
         return "-"
     if value is None:
         return "-"
     return value
 
 
+def clean_tags(text):
+    """Sanitizes ALL HTML tags that are not <mark>"""
+    allowed_tags = ["mark"]
+    return bleach.clean(text, tags=allowed_tags)
```

Comment:
this still is dangerous if someone has put malicious code in a "mark" tag,
e.g. `<mark onclick="alert('XSS')">Click here</mark>`

So I think the suggestion of using a uuid as original tag that we want opensearch to wrap our results in, then doing this clean with no allowed_tags and then str replacing it with mark afterwards, would be the only way to be 100% safe.

Let's chat through it though